src/types.ts

```diff
@@ -14,76 +15,86 @@ export type InvocationPayload<Body extends ValidInvocationPayloadBody> = {
     function: {
       callback_id: string;
     };
     function_execution_id: string;
-    inputs: FunctionOutputInputValues;
+    inputs: FunctionInputValues;
+    bot_access_token?: string;
   };
 };
 
+export type BlockActionInvocationBody = {
+  type: "block_actions";
+  actions: BlockAction[];
+  bot_access_token?: string;
+  function_data?: FunctionData;
+  // deno-lint-ignore no-explicit-any
+  [key: string]: any;
+};
+
+type FunctionData = {
+  function: {
+    callback_id: string;
+  };
+  execution_id: string;
+  inputs: FunctionInputValues;
+};
+
 // TODO: type this to account for variable return options?
 export type FunctionHandlerReturnArgs = {
   completed?: boolean;
-  outputs?: FunctionOutputInputValues;
+  outputs?: FunctionOutputValues;
   error?: string;
 };
 
-export type FunctionContext = {
+export type FunctionHandlerArgs = {
   env: EnvironmentVariables;
-  inputs: FunctionOutputInputValues;
+  inputs: FunctionInputValues;
   token: string;
   event: FunctionInvocationBody["event"];
 };
 
-export type AsyncFunctionHandler = {
-  (context: FunctionContext): Promise<FunctionHandlerReturnArgs>;
+export type FunctionHandler = {
+  (
+    args: FunctionHandlerArgs,
+  ): Promise<FunctionHandlerReturnArgs> | FunctionHandlerReturnArgs;
 };
 
-export type SyncFunctionHandler = {
-  (context: FunctionContext): FunctionHandlerReturnArgs;
-};
-
-export type FunctionHandler = AsyncFunctionHandler | SyncFunctionHandler;
-
 // This is the interface a developer-provided function module should adhere to
 export type FunctionModule = {
   default: FunctionHandler;
+  blockActions?: BlockActionsHandler;
 };
 
-export type BaseResponse = {
-  /** `true` if the response from the server was successful, `false` otherwise. */
-  ok: boolean;
-  /** Optional error description returned by the server. */
-  error?: string;
-  /** Optional list of warnings returned by the server. */
-  warnings?: string[];
-  /** Optional metadata about the response returned by the server. */
-  response_metadata?: {
-    warnings?: string[];
-    messages?: string[];
-  };
-  [otherOptions: string]: unknown;
-};
-
-export interface ISlackAPIClient {
-  /**
-   * Calls a Slack API method.
-   * @param {string} method The API method name to invoke, i.e. `chat.postMessage`.
-   * @param {Object} data Object representing the data you wish to send along to the Slack API method.
-   * @returns {Promise<BaseResponse>} A Promise that resolves to the data the API responded with.
-   * @throws {Error} Throws an Error if the API response was not OK or a network error occurred.
-   */
-  call(method: string, data: { [key: string]: unknown }): Promise<BaseResponse>;
-}
-
 export const EventTypes = {
   FUNCTION_EXECUTED: "function_executed",
+  BLOCK_ACTIONS: "block_actions",
 } as const;
 
 export type ValidEventType = typeof EventTypes[keyof typeof EventTypes];
+
+// --- Block Actions Types -- //
+// deno-lint-ignore no-explicit-any
+type BlockAction = any;
```

Comment:
Do we have any indication on what this might look like? Is it always different or is it just in flux atm?

Reply:
Yes, it looks like [this](https://github.com/slackapi/bolt-js/blob/main/src/types/actions/block-action.ts#L210) - typed in bolt js. The reason I chose `any` here was that for the purposes of the runtime, it doesn't care what it looks like beyond just being present as an action in some payloads. I could see eventually just having this runtime library use a separate slack types library though for these payloads.